tests/main/apparmor-prompting-flag-restart/task.yaml

```diff
@@ -0,0 +1,121 @@
+summary: Check that changing the experimental.apparmor-prompting flag causes snapd to restart
+
+details: |
+    This test checks that snapd is restarted when the experimental.apparmor-prompting
+    flag is changed, and that snapd is not restarted when the flag is set to
+    the same value repeatedly. Restarting snapd when the flag changes is
+    necessary so that apparmor kernel and parser features are re-probed, in
+    case support for prompting has changed since snapd last started.
+
+systems:
+  - ubuntu-16.04-*
+  - ubuntu-18.04-*
+  - ubuntu-2*
+
+prepare: |
+    if ! command -v curl; then
+        snap install --devmode --edge test-snapd-curl
+        snap alias test-snapd-curl.curl curl
+    fi
+
+debug: |
+    # Report system-info
+    curl -sS --unix-socket /run/snapd.socket http://localhost/v2/system-info | jq
+
+execute: |
+    . /etc/os-release
+
+    # Necessary since we restart snapd many times
+    restart_snapd() {
+        systemctl stop snapd.service snapd.socket
+        systemctl reset-failed snapd.service snapd.socket
+        systemctl start snapd.service
+        retry --wait 1 -n 100 sh -x -c 'systemctl is-active snapd.service snapd.socket'
+    }
+
+    # Check that snapd has restarted and now has a new PID, then set SNAPD_PID to that new PID.
+    check_snapd_restarted() {
+        #shellcheck disable=SC2016
+        retry --wait 1 -n 100 --env SNAPD_PID="$SNAPD_PID" sh -x -c 'systemctl is-active snapd.service snapd.socket && test ! "$(systemctl show --property MainPID snapd.service | cut -f2 -d=)" = "$SNAPD_PID"'
+        # Set SNAPD_PID so future checks use most recent PID
+        SNAPD_PID="$(systemctl show --property MainPID snapd.service | cut -f2 -d=)"
```

Comment:
there's also InvocationID, which is a random string generated by systemd, but I don't recall how far back this feature dates in systemd.

Reply:
It certainly predates any prompting-related patch in apparmor, so would probably be fine, but I don't see it used in any existing tests, so I'm inclined to stick with `MainPID` here. And when a proper fix within snapd is ready, this should be removed anyway.